tests/kernel/ib_tests.pm

```diff
@@ -0,0 +1,207 @@
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$host2");
+    script_run("/usr/bin/clear");
+
+    barrier_wait('IBTEST_SETUP');
+    # wait until the two machines under test are ready setting up their local things
+    barrier_wait('IBTEST_BEGIN');
+
+    assert_script_run('cd hpc-testing');
+    assert_script_run("./ib-test.sh $host1 $host2", 1800);
+
+
+    # remove non-printable characters
+    script_run('tr -cd \'\11\12\15\40-\176\' < results/TEST-ib-test.xml > /tmp/results.xml');
+    parse_extra_log('XUnit', '/tmp/results.xml');
+
+    script_run("scp -o StrictHostKeyChecking=no root\@$host1:/tmp/dmesg.txt /tmp/dmesg_slave1.txt");
+    script_run("scp -o StrictHostKeyChecking=no root\@$host2:/tmp/dmesg.txt /tmp/dmesg_slave2.txt");
+
+    check_dmesg('/tmp/dmesg_slave1.txt');
+    check_dmesg('/tmp/dmesg_slave2.txt');
+
+    barrier_wait('IBTEST_DONE');
+    barrier_destroy('IBTEST_SETUP');
+    barrier_destroy('IBTEST_BEGIN');
+    barrier_destroy('IBTEST_DONE');
+}
+
+sub run {
+    my $role = get_required_var('IBTEST_ROLE');
+
+    select_console 'root-ssh';
+    zypper_call('ar -f -G ' . get_required_var('GA_REPO'));
+
+    if ($role eq "IBTEST_MASTER") {
+        zypper_call('ar -f -G ' . get_required_var('DEVEL_TOOLS_REPO'));
+    }
+
+    zypper_call('--gpg-auto-import-keys ref');
+
+    if ($role eq "IBTEST_MASTER") {
+        ibtest_master;
+    }
+    elsif ($role eq 'IBTEST_SLAVE1' || $role eq 'IBTEST_SLAVE2') {
+        ibtest_slave;
+    }
+
+    power_action('poweroff', observe => 0, keepconsole => 0);
+}
+
+sub test_flags {
+    return {fatal => 1};
+}
+
+sub post_fail_hook {
+    script_run('tr -cd \'\11\12\15\40-\176\' < results/TEST-ib-test.xml > /tmp/results.xml');
+    parse_extra_log('XUnit', '/tmp/results.xml');
+}
+
+1;
+
+=head1 bare metal testing for InfiniBand
+
+=head2 Overview
+This test is executing the InfiniBand testsuite currently under development at
+https://gitlab.suse.de/NMoreyChaisemartin/hpc-testing/
+
+In order to run this testsuite, three IPMI-workers are required. Two of them need 
+to have actual InfiniBand HCA's, the third one is controlling the test execution.
+
+The test has some additional dependencies. The controlling master needs to have "twopence"
+(see https://github.com/openSUSE/twopence) which is not in the default SLE repositories. 
+A repository needs to be specified
+
+=head1 openQA setup
+
+=head2 openQA worker setup
+The workers with the InfiniBand HCA's need a special worker class, in this case
+we assume it is "64bit-ipmi_infiniband". The third worker just needs a different 
+worker class, something like "64bit-ipmi" should be fine.
+
+=head2 openQA test suites
+As the test is executed on three hosts, three test suites should be created:
+
+=head3 ibtest-master	
+ADDONURL_SDK=ftp://openqa.suse.de/SLE-15-SDK-POOL-%ARCH%-Build%BUILD%-Media1/
+GA_REPO="http://download.suse.de/ibs/SUSE:/SLE-15:/GA/standard/SUSE:SLE-15:GA.repo"
```

Comment:
Not sure if everybody is fine with having internal URLs in public repos. I'd repalce it with an example URL but no hard feelings

Reply:
Valid point. I'll change it.

Reply:
Or simply follow "Factory first" and provide primarily tests that work for openSUSE Tumbleweed. Then you can easily use public OBS repos as reference anyway :)

Reply:
Unfortunately, factory first will not work here, as the machines required for this testsuite are not accessible from openqa.opensuse.org